overlord/servicestate/service_control.go

```diff
@@ -0,0 +1,124 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package servicestate
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/overlord/snapstate"
+	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/wrappers"
+
+	tomb "gopkg.in/tomb.v2"
+)
+
+// ServiceAction encapsulates a single service-related action (such as starting,
+// stopping or restarting) run against services of a given snap. The action is
+// run for services listed in names attribute, or for all services of the snap
+// if names is empty.
+type ServiceAction struct {
+	SnapName       string                 `json:"snap-name"`
+	Action         string                 `json:"action"`
+	ActionModifier string                 `json:"action-modifier,omitempty"`
+	Services       []string               `json:"names,omitempty"`
+	StopReason     snap.ServiceStopReason `json:"stop-reason,omitempty"`
```

Comment:
Could stop reason be generalized to just reason? It could be unset if there's no specific reason but I think having a distinct stop reason on a general action struct is weird.

Reply:
after discussing we don't need a StopReason at all there

Reply:
Removed.